documentation/src/main/asciidoc/user_guide/locking.adoc

```diff
@@ -29,112 +36,101 @@ NOTE: If the operation is conditional and it fails on the primary owner, it is n
+----
 
-This scheme is used to encapsulate an external source of data versioning within Infinispan, such as when using Infinispan with Hibernate which in turn gets its data version information directly from a database.
+Or
 
-In this scheme, a mechanism to pass in the version becomes necessary, and overloaded versions of put() and putForExternalRead() will be provided in AdvancedCache to take in an external data version.  This is then stored on the InvocationContext and applied to the entry at commit time. 
+[source,java]
+----
+new ConfigurationBuilder().locking().concurrencyLevel(32);
+----
 
-Write skew checks cannot and will not be performed in the case of external data versioning.
+==== Lock timeout
 
-==== Tombstones
-To deal with deletions of entries, tombstones will be maintained as null entries that have been deleted, so that version information of the deleted entry can be maintained and write skews can still be detected.  However this is an expensive thing to do, and as such, is a configuration option, disabled by default. Further, tombstones will follow a strict lifespan and will be cleared from the system after a specific amount of time. 
+The lock timeout specifies the amount of time, in milliseconds, to wait for a contented lock.
 
-==== Configuration
-By default versioning will be _disabled_.  This will mean write skew checks when using transactions and _REPEATABLE_READ_ as an isolation level will be unreliable when used in a cluster.
-Note that this doesn't affect single-node, LOCAL mode usage. 
+*Configuration example:*
 
 [source,xml]
 ----
-
-<versioning scheme="SIMPLE|NONE" />
-
+<locking acquire-timeout="10000"/>
 ----
 
 Or
 
 [source,java]
 ----
+new ConfigurationBuilder().locking().lockAcquisitionTimeout(10000);
+//alternatively
+new ConfigurationBuilder().locking().lockAcquisitionTimeout(10, TimeUnit.SECONDS);
+----
 
-new ConfigurationBuilder().versioning().scheme(SIMPLE);
+==== Consistency
+The fact that a single owner is locked (as opposed to all owners being locked) does not break the following consistency guarantee:
+if key `K` is hashed to nodes `{A, B}` and transaction `TX1` acquires a lock for `K`, let's say on `A`.
+If another transaction, `TX2`, is started on `B` (or any other node) and `TX2` tries to lock `K` then it will fail with a timeout as the lock is already held by `TX1`.
+The reason for this is the that the lock for a key `K` is always, deterministically, acquired on the same node of the cluster, regardless of where the transaction originates.
 
-----
+===  Data Versioning
+Infinispan supports two forms of data versioning: simple and external.
+The simple versioning is used in transactional caches for write skew check.
+Check section <<tx:write-skew>> section for detail about it.
+
+The external versioning is used to encapsulate an external source of data versioning within Infinispan, such as when using Infinispan with Hibernate which in turn gets its data version information directly from a database.
```

Comment:
it doesn't use it. 
it only says that the user can store its own version (via Metadata). from that, it can install custom interceptors to manage it if needed.